core/test/unit/cdk-nag/nag-data-domain-crawler.test.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+/**
+ * Tests DataDomainCrawler
+ *
+ * @group unit/best-practice/data-domain-crawler
+ */
+
+import { Annotations, Match } from 'aws-cdk-lib/assertions';
+import { Role, CompositePrincipal, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { EventBus } from 'aws-cdk-lib/aws-events';
+import { App, Stack, Aspects, RemovalPolicy, Aws } from 'aws-cdk-lib';
+// eslint-disable-next-line import/no-extraneous-dependencies,import/no-unresolved
+import { AwsSolutionsChecks, NagSuppressions } from 'cdk-nag';
+import { DataDomainCrawler } from '../../../src/data-mesh'
+
+
+const mockApp = new App();
+
+const dataDomainCrawlerStack = new Stack(mockApp, 'dataDomainCrawler');
+
+const lfAdminRole = new Role(dataDomainCrawlerStack, 'myRole', {
+  assumedBy: new CompositePrincipal(
+    new ServicePrincipal('glue.amazonaws.com'),
+    new ServicePrincipal('lakeformation.amazonaws.com'),
+    new ServicePrincipal('states.amazonaws.com')
+  ),
+});
+lfAdminRole.applyRemovalPolicy(RemovalPolicy.DESTROY)
+
+const eventBus = new EventBus(dataDomainCrawlerStack, 'dataDomainEventBus', {
+  eventBusName: `${Aws.ACCOUNT_ID}_dataDomainEventBus`,
+});
+eventBus.applyRemovalPolicy(RemovalPolicy.DESTROY);
+
+new DataDomainCrawler(dataDomainCrawlerStack, 'myDataDomainCrawler', {
+  lfAdminRole,
+  eventBus,
+  dataDomainWorkflowArn: "arn-test"
+});
+
+Aspects.of(dataDomainCrawlerStack).add(new AwsSolutionsChecks());
+
+NagSuppressions.addResourceSuppressionsByPath(
```

Comment:
the role should be split into multiple purpose built roles with narrow permissions. The role is getting inline policies so it should be tested

Reply:
will be split in another version